src/_P039_Thermosensors.ino

```diff
@@ -0,0 +1,1681 @@
+      break;
+    }
+
+    case PLUGIN_READ:
+    {
+      // Get the MAX Type (6675 / 31855 / 31856)
+      byte MaxType = P039_MAX_TYPE;
+
+      float Plugin_039_Celsius = NAN;
+
+      switch (MaxType) {
+        case P039_MAX_6675:
+          Plugin_039_Celsius = readMax6675(event);
+          break;
+        case P039_MAX_31855:
+          Plugin_039_Celsius = readMax31855(event);
+          break;
+        case P039_MAX_31856:
+          Plugin_039_Celsius = readMax31856(event);
+          break;
+        case P039_MAX31865:
+          Plugin_039_Celsius = readMax31865(event);
+          break;
+        case P039_LM7x:
+          Plugin_039_Celsius = readLM7x(event);
+          break;
+      }
+
+      if (Plugin_039_Celsius != NAN)
+      {
+        UserVar[event->BaseVarIndex] = Plugin_039_Celsius;
+
+        if (loglevelActiveFor(LOG_LEVEL_INFO)) {
+          String log;
+          if((log.reserve(66u))) {
+            log = F("P039 : ");
+            log += getTaskDeviceName(event->TaskIndex);
+            log += F(" : ");
+          
+            for (uint8_t i = 0; i < getValueCountForTask(event->TaskIndex);i++)
+            {
+              log += getTaskValueName(event->TaskIndex, i);
+              log += F(": ");
+              log += formatUserVarNoCheck(event->TaskIndex, i); 
+              log += ' '; 
+            }
+            addLog(LOG_LEVEL_INFO, log);
+          }
+        }
+        success = true;
+      }
+      else
+      {
+        String log;
+        if((log.reserve(66u))) {
+          log = F("P039 :  ");
+          log += getTaskDeviceName(event->TaskIndex);
+          log += F(" : ");
+          log += F("No Sensor attached !");
+          UserVar[event->BaseVarIndex]     = NAN;
+          UserVar[event->BaseVarIndex + 1] = NAN;
+          addLog(LOG_LEVEL_INFO, log);
+        }
+        success = false;
+      }
+
+      break;
+    }
+  }
+  return success;
+}
+
+float readMax6675(struct EventStruct *event)
+{
+  uint8_t CS_pin_no = get_SPI_CS_Pin(event);
+
+  // "transfer" 0x0 and read the Data from the Chip
+  // uint16_t rawvalue = SPI.transfer16(0x0);
+  uint16_t rawvalue = read16BitRegister(CS_pin_no, 0x0000);
+
+  # ifndef BUILD_NO_DEBUG
+
+    if (loglevelActiveFor(LOG_LEVEL_DEBUG)) 
+    {
+      String log;
+      if((log.reserve(66u))) {
+        log = F("P039 : MAX6675 : RAW - BIN:");
+        log += String(rawvalue, BIN);
+        log += F(" HEX:");
+        log += String(rawvalue, HEX);
+        log += F(" DEC:");
+        log += String(rawvalue);
+        addLog(LOG_LEVEL_DEBUG, log);
+      }
+    }
+
+  # endif // ifndef BUILD_NO_DEBUG
+
+  // Open Thermocouple
+  // Bit D2 is normally low and goes high if the thermocouple input is open. In order to allow the operation of the
+  // open  thermocouple  detector,  T-  must  be  grounded. Make  the  ground  connection  as  close  to  the  GND  pin
+  // as possible.
+  const bool Plugin_039_SensorAttached = !(rawvalue & 0x0004);
```

Comment:
@chri-kai-in but D1 is low always as 6675 ID, so we can use it also here.

Reply:
Fine for a re-work of the thermocouple sections. Initially i was not planning to "crash the front door" on those lines of code . Without hardware for testing i felt not in the position to touch those, except the communication handlers.

I will add this to my list.

Reply:
FIXED: reworked to flag defines in latest push